GNUmakefile

```diff
@@ -321,6 +322,10 @@ lint-tools:
 proto-tools:
 	@$(SHELL) $(CURDIR)/build-support/scripts/devtools.sh -protobuf
 
+.PHONY: codegen-tools
```

Comment:
I think we need a `make` target to correctly re-execute all of the codegen. Ideally also do something in CI that fails if a fresh run of the deep-copy generators produces a diff (like the `go.mod/go.sum` divergence checks).

Reply:
Sounds good. Would something like this do the trick?

```
generate-deep-copy:
	go generate ./agent/structs ./proto/pbpeering ./agent/proxycfg
```

That'd technically run any other `go:generate` directives in those packages, so would it be better to run the `deep-copy.sh` scripts directly?

Reply:
Added a make target and CI check ✅ (went with the latter approach)